src/components/Modals/Send/utils.ts

```diff
@@ -315,3 +304,47 @@ export const handleSend = async ({
 
   return broadcastTXID
 }
+
+const prepareSendAdapter = (sendInput: SendInput) => {
+  const state = store.getState()
+  const asset = selectAssetById(state, sendInput.assetId ?? '')
+  if (!asset) throw new Error(`No asset found for assetId ${sendInput.assetId}`)
+
+  const chainId = asset.chainId
+  const accountMetadata = selectPortfolioAccountMetadataByAccountId(state, {
+    accountId: sendInput.accountId,
+  })
+  if (!accountMetadata) {
+    throw new Error(`No accountMetadata found for ${sendInput.accountId}`)
+  }
+
+  const adapter = assertGetChainAdapter(chainId)
+
+  return { asset, chainId, accountMetadata, adapter }
+}
+
+export const maybeFetchChangeAddress = async ({
+  sendInput,
+  wallet,
+}: {
+  sendInput: SendInput
+  wallet: HDWallet
+}): Promise<string | undefined> => {
+  try {
+    const { chainId, accountMetadata, adapter } = prepareSendAdapter(sendInput)
+
+    // Only fetch for UTXO chains on Ledger wallets
+    if (!isUtxoChainId(chainId) || !isLedger(wallet)) return undefined
+
+    const changeAddress = await adapter.getAddress({
+      accountNumber: accountMetadata.bip44Params.accountNumber,
+      accountType: accountMetadata.accountType,
+      wallet,
+      isChange: true,
```

Comment:
preferably-blocking: this may accidentally work for sends since we don't have an *explicit* change output, but if the intent is to bring this flow across the board, then we run the risk of displaying the wrong change address:

Say my change index is 10, but for any reason, I'm sending an UTXO with e.g 0 as a change address (say, for THORFi things where we *only* pick UTXOs for index 0 for inputs, and enforce index 0 change address IIRC) then that could be an issue.

Not a big deal for the time being, since that's constrained to sends, but ideally, we would be able to introspect the unsigned Tx change output to ensure we always display the right change address.